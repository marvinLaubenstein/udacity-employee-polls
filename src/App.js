import React from 'react';
import { useEffect, Fragment } from 'react';
import { connect } from 'react-redux';
import { handleInitialData } from './actions/shared';
import Dashboard from './components/dashboard/Dashboard';
import QuestionPollPage from './components/question-poll-page/QuestionPollPage';
import QuestionAddingPage from './components/question-adding-page/QuestionAddingPage';
import LoadingBar from 'react-redux-loading-bar';
import LoginScreen from './components/login/LoginScreen';
import { Routes, Route } from 'react-router-dom';
import LeaderBoard from './components/leader-board/LeaderBoard';
import PrivateRoutes from './components/privateroute/PrivateRoutes';

const App = ({ dispatch, isLoggedIn, loading }) => {
  useEffect(() => {
    dispatch(handleInitialData());
  });
  return (
    <Fragment>
      <LoadingBar />
      <div className="container">
        {loading === true ? null : (
          <Routes>
            <Route path="/login" exact element={<LoginScreen />}></Route>
            <Route element={<PrivateRoutes />}>
              <Route path="/" exact element={<Dashboard />} />
              <Route
                path="/questions/:id"
                exact
                element={<QuestionPollPage />}
              />
              <Route path="/add" exact element={<QuestionAddingPage />} />
              <Route
                path="/leaderboard"
                exact
                element={<LeaderBoard />}
              ></Route>
            </Route>
            <Route
              path="*"
              element={<p>There's nothing here: 404 Error (Page Not Found)!</p>}
            />
          </Routes>
        )}
      </div>
    </Fragment>
  );
};

const mapStateToProps = ({ authedUser }) => ({
  loading: !!authedUser ? authedUser === null : false,
  isLoggedIn: !!authedUser,
});

export default connect(mapStateToProps)(App);
