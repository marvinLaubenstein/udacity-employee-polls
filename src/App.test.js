import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import reducer from './reducers';
import middleware from './middleware';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

describe('App', () => {
  let component;
  const store = createStore(reducer, middleware);
  beforeEach(async () => {
    component = render(
      <Provider store={store}>
        <BrowserRouter>
          <App />
        </BrowserRouter>
      </Provider>
    );
  });
  it('should match snapshot after render without login', () => {
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
  it('should render Loginscreen', () => {
    expect(component.getByTestId('login')).toBeInTheDocument();
  });
});
