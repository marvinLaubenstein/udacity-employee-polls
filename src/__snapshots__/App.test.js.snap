// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot after render without login 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="direction: ltr;"
      >
        <div
          class=""
          style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
        />
        <div
          style="display: table; clear: both;"
        />
      </div>
      <div
        class="container"
      >
        <div
          class="MuiGrid-root"
          data-testid="login"
        >
          <div
            class="MuiPaper-root login-screen-paper MuiPaper-elevation10 MuiPaper-rounded"
            style="padding: 20px; width: 280px; margin: 20px auto;"
          >
            <div
              align="center"
              class="MuiGrid-root"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
                style="background-color: rgb(4, 170, 109);"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                  />
                </svg>
              </div>
              <h2>
                Welcome-Login
              </h2>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              style="padding-bottom: 10px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
              >
                Username
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  placeholder="Enter username"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      Username
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
              style="padding-bottom: 10px;"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                data-shrink="false"
              >
                Password
                <span
                  aria-hidden="true"
                  class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                >
                   
                  *
                </span>
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  placeholder="Enter password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      Password
                       *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
              style="padding: 10px; background: rgb(4, 170, 109);"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Login
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="direction: ltr;"
    >
      <div
        class=""
        style="width: 0%; transition: width 800ms linear 0s; will-change: width, opacity; height: 3px; background-color: red; position: absolute; opacity: 0;"
      />
      <div
        style="display: table; clear: both;"
      />
    </div>
    <div
      class="container"
    >
      <div
        class="MuiGrid-root"
        data-testid="login"
      >
        <div
          class="MuiPaper-root login-screen-paper MuiPaper-elevation10 MuiPaper-rounded"
          style="padding: 20px; width: 280px; margin: 20px auto;"
        >
          <div
            align="center"
            class="MuiGrid-root"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
              style="background-color: rgb(4, 170, 109);"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
                />
              </svg>
            </div>
            <h2>
              Welcome-Login
            </h2>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            style="padding-bottom: 10px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
            >
              Username
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                placeholder="Enter username"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Username
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            style="padding-bottom: 10px;"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
            >
              Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input"
                placeholder="Enter password"
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-3"
                >
                  <span>
                    Password
                     *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
            style="padding: 10px; background: rgb(4, 170, 109);"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Login
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
